pkg/neotest/basic.go

```diff
@@ -297,13 +298,45 @@ func AddSystemFee(bc *core.Blockchain, tx *transaction.Transaction, sysFee int64
 }
 
 // AddNetworkFee adds network fee to the transaction.
-func AddNetworkFee(bc *core.Blockchain, tx *transaction.Transaction, signers ...Signer) {
+func AddNetworkFee(t testing.TB, bc *core.Blockchain, tx *transaction.Transaction, signers ...Signer) {
 	baseFee := bc.GetBaseExecFee()
 	size := io.GetVarSize(tx)
 	for _, sgr := range signers {
-		netFee, sizeDelta := fee.Calculate(baseFee, sgr.Script())
-		tx.NetworkFee += netFee
-		size += sizeDelta
+		if csgr, ok := sgr.(ContractSigner); ok {
+			// from https://github.com/nspcc-dev/neo-go/blob/v0.103.1/pkg/core/blockchain.go#L2814
+
+			require.NoError(t, vm.IsScriptCorrect(csgr.InvocationScript(), nil))
+
+			cs := bc.GetContractState(csgr.ScriptHash())
```

Comment:
To be honest, I don't like this code duplication. If we're going the "invocation" way, then why not to use the approach similar to `invokecontractverify` RPC call?
```
		ic, err := bc.GetTestVM(trigger.Verification, tx, nil)
		require.NoError(t, err)
		bc.InitVerificationContext(ic, sgr.ScriptHash(), &transaction.Witness{InvocationScript: csgr.InvocationScript(), VerificationScript: []byte{}})
```
Likely it would be better to create a copy of `tx` in order not to cache its has, because some interops may call `tx.Hash()`.
Take a look at the https://github.com/nspcc-dev/neo-go/blob/fd7b7ffc134481bf639ed8daf451a4b6b6f990be/pkg/services/rpcsrv/server.go#L2249.

Another approach I've considered is based on the `calculatenetworkfee` RPC call handler, but it's probably less precise, because here we have an exact invocation script, and RPC handler relies only to `verify`'s parameters.

Reply:
> RPC handler relies only to verify's parameters

If invocation script is not provided, but if it is --- no magic, the script is used as is.

Reply:
Fixed.